File: app/src/main/java/ir/beigirad/zeroapplication/ui/MainActivity.java
package ir.beigirad.zeroapplication.ui;

import android.os.Bundle;

import ir.beigirad.zeroapplication.R;
import ir.beigirad.zeroapplication.component.BaseActivity;

public class MainActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);


    }

    @Override
    protected int getContentView() {
        return R.layout.activity_main;
    }

    @Override
    protected CharSequence getToolbarTitle() {
        return "MainActivity";
    }

}

File: app/src/main/java/ir/beigirad/app/MainActivity.kt
package ir.beigirad.app

import androidx.appcompat.widget.Toolbar
import ir.beigirad.zeroapplication.bases.BaseActivity
import kotlinx.android.synthetic.main.app_bar.*

/**
 * Created by farhad-mbp on 3/18/18.
 */
class MainActivity : BaseActivity() {
    override val contentView: Int
        get() = R.layout.activity_main
    override val toolbar: Toolbar?
        get() = app_toolbar

    override val toolbarTitle: Int?
        get() = R.string.app_title
    override val hasBackConfirmation: Boolean
        get() = true

    override fun initToolbar(hasBack: Boolean) {
        super.initToolbar(hasBack = false)
    }


}